### PR TITLE
SW-1226-change-the-cloud-config-to-use-beamos-version-instead-of-date

### PR DESCRIPTION
**Pull request by JFSolution**
change the cloud config to use beamos version instead of beamos_date
- refactor beamos_date to beamos_version
- changed comparision to use the version instead of the date

### CODE REVIEW

File: octoprint_mrbeam/software_update_information.py [hunk 1]
```diff
@@ -265,8 +267,8 @@ def _set_info_from_cloud_config(plugin, tier, beamos_date, cloud_config):
                 <module_id>: {
                     <module_settings>,
                     <tier>:{<tier_settings>},
-                    "beamos_date": {
-                        <YYYY-MM-DD>: {<beamos_settings>}
+                    "beamos_version": {
+                        "X.X.X": {<beamos_settings>}
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:271]: So, we only support base version ? what if the version is following PEP440 as well ?

File: octoprint_mrbeam/software_update_information.py [hunk 2]
```diff
@@ -479,35 +481,36 @@ def _get_curent_version(input_moduleconfig, module_id, plugin):
...
     """
-    if "beamos_date" not in moduleconfig:
+    if "beamos_version" not in moduleconfig:
         return {}
 
-    beamos_date_config = {}
-    prev_beamos_date_entry = datetime.strptime("2000-01-01", "%Y-%m-%d").date()
-    for date, beamos_config in moduleconfig["beamos_date"].items():
+    beamos_version_config = {}
+    # TODO change to version instead of date
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:499]: not needed anymore ?

File: octoprint_mrbeam/software_update_information.py [hunk 3]
```diff
@@ -479,35 +481,36 @@ def _get_curent_version(input_moduleconfig, module_id, plugin):
...
+    prev_beamos_version_entry = "0.0.0"
+    for beamos_version, beamos_config in moduleconfig["beamos_version"].items():
         if (
-            beamos_date >= datetime.strptime(date, "%Y-%m-%d").date()
-            and prev_beamos_date_entry < beamos_date
+            version.parse(beamos_version) >= version.parse(beamos_version)
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:503]: this is the same variable in both sides right ?

File: octoprint_mrbeam/software_update_information.py [hunk 4]
```diff
@@ -479,35 +481,36 @@ def _get_curent_version(input_moduleconfig, module_id, plugin):
...
-    generates the config for the given beamos_date of the tierversion
+    generates the config for the given beamos_version of the tierversion
     Args:
         moduleconfig: update config of the module
-        beamos_date: date of the beamos
+        beamos_version: version of the beamos
         tierversion: software tier
 
     Returns:
         beamos config of the tierversion
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:493]: missing the return type

File: octoprint_mrbeam/software_update_information.py [hunk 5]
```diff
@@ -479,35 +481,36 @@ def _get_curent_version(input_moduleconfig, module_id, plugin):
     return current_version
 
 
-def _generate_config_of_beamos(moduleconfig, beamos_date, tierversion):
+def _generate_config_of_beamos(moduleconfig, beamos_version, tierversion):
     """
-    generates the config for the given beamos_date of the tierversion
+    generates the config for the given beamos_version of the tierversion
     Args:
         moduleconfig: update config of the module
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:488]: missing the type

File: octoprint_mrbeam/software_update_information.py [hunk 6]
```diff
@@ -479,35 +481,36 @@ def _get_curent_version(input_moduleconfig, module_id, plugin):
...
-        beamos_date: date of the beamos
+        beamos_version: version of the beamos
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:489]: missing the type

File: octoprint_mrbeam/software_update_information.py [hunk 7]
```diff
@@ -479,35 +481,36 @@ def _get_curent_version(input_moduleconfig, module_id, plugin):
...
         tierversion: software tier
 
     Returns:
         beamos config of the tierversion
     """
-    if "beamos_date" not in moduleconfig:
+    if "beamos_version" not in moduleconfig:
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:495]: Probably it would be good if we add a debug log  if we enter the if

**Review by ahmed-mrbeam — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/software_update_information.py (hunk 1 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:271]: add comment that the version follow PEP-440

File: octoprint_mrbeam/software_update_information.py (hunk 2 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:499]: you are right, will remove this

File: octoprint_mrbeam/software_update_information.py (hunk 3 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:503]: You are right I have to check

File: octoprint_mrbeam/software_update_information.py (hunk 4 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:493]: will add

File: octoprint_mrbeam/software_update_information.py (hunk 5 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:488]: same as before, will add

File: octoprint_mrbeam/software_update_information.py (hunk 7 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:495]: I can add one

File: octoprint_mrbeam/software_update_information.py (hunk 1 above)
Comment (ahmed-mrbeam) [reply octoprint_mrbeam/software_update_information.py:271]: where is the comment ?

Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:271]: in this PR https://github.com/mrbeam/MrBeamPlugin/pull/1458

**Comment by JFSolution** [comment]
closed as the changes were merged with https://github.com/mrbeam/MrBeamPlugin/pull/1458

**Pull request closed without merging**